Packages/Tracking/Core/Runtime/Scripts/LeapServiceProvider.cs

```diff
@@ -647,6 +647,11 @@ protected virtual void FixedUpdate()
             else
             {
                 _leapController.Frame(_untransformedFixedFrame);
+
+                if (_currentDevice != null)
+                {
+                    _untransformedFixedFrame.DeviceID = _currentDevice.DeviceID;
+                }
```

Comment:
I think this might actually draw attention to a larger issue

Conection.Frames has no idea about devices, and so is a stack of _all_ frames, regardless of device. We then populate it with whichever frame is available

This means when interpolation is not enabled, the provided data can be from any active device, regardless of the device that should be used for the specific LeapServiceProvider